### PR TITLE
Transform mouse events to the local coordinate system

### PR DESCRIPTION
**Pull request by dkwingsmt**
## Description

This PR fixes the issue that mouse events from callbacks (`onEnter`, `onHover`, `onExit`) don't have correct local information, i.e. `localPosition` and `localDelta`.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from our [issue database]. Indicate, which of these issues are resolved or fixed by this PR. There should be at least one issue listed here.*

## Tests

I added the following tests:

*Replace this with a list of the tests that you added as part of this PR. A change in behaviour with no test covering it
will likely get reverted accidentally sooner or later. PRs must include tests for all changed/updated/fixed behaviors. See [Test Coverage].*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I signed the [CLA].
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [ ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ ] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by goderbauer** [comment]
/cc @gspencergoog 
Looks like we have to PRs that implement the same feature? See https://github.com/flutter/flutter/pull/48271.

File: packages/flutter/lib/src/rendering/proxy_box.dart [hunk 1]
```diff
@@ -2750,6 +2750,10 @@ class RenderMouseRegion extends RenderProxyBox {
   @visibleForTesting
   MouseTrackerAnnotation get hoverAnnotation => _hoverAnnotation;
 
+  Matrix4 _eventTransformer() {
+    return Matrix4.inverted(getTransformTo(null));
```
Comment (goderbauer) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: Same comment here that I also left on https://github.com/flutter/flutter/pull/48271

**Comment by kenthinson** [comment]
Hi I need this for a project I'm working on. Any ETA for a merge into master? Thanks for the hard work :)

**Comment by dkwingsmt** [comment]
Closing this PR due to performance issue mentioned in https://github.com/flutter/flutter/pull/48271#discussion_r363885281. We do have plans to implement this feature in other ways.

**Pull request closed without merging**